src/transformStateWithClones.js

```diff
@@ -6,8 +6,31 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
+
+  actions.forEach((action) => {
+    const baseS = arrState.length > 0 ? arrState[arrState.length - 1] : state;
```

Comment:
The logic for determining `baseS` is correct, but it could be simplified by initializing `arrState` with `[state]` and then always using the last element of `arrState` as `baseS`.